core/crdt/pncounter.go

```diff
@@ -108,13 +114,30 @@ func (reg PNCounter[T]) Value(ctx context.Context) ([]byte, error) {
 }
 
 // Set generates a new delta with the supplied value
-func (reg PNCounter[T]) Increment(value T) *PNCounterDelta[T] {
+func (reg PNCounter[T]) Increment(ctx context.Context, value T) (*PNCounterDelta[T], error) {
+	// To ensure that the dag block is unique, we add a random number to the delta.
+	// This is done only on update (if the doc doesn't already exist) to ensure that the
+	// initial dag block of a document can be reproducible.
+	exists, err := reg.store.Has(ctx, reg.key.ToPrimaryDataStoreKey().ToDS())
+	if err != nil {
+		return nil, err
+	}
+	var entropy int64
+	if exists {
```

Comment:
question: When creating a new PNCounter field, this is going to produce an entropy value of `0`. Is this intended? Or do we want to omit the entropy field entirely when creating? (IE: not include in the cbor marshal)

Reply:
This is indeed intended. Instead of having a different struct for create vs update, using 0 gives us the same result with less complexity.

Reply:
A clarification. The entropy will be `0` only when creating the **document**.